addons/cert.py

```diff
@@ -157,6 +157,17 @@ def exp42(data):
                 token, 'style', "Comparison of struct padding data " +
                 "(fix either by packing the struct using '#pragma pack' or by rewriting the comparison)", 'EXP42-C')
 
+# EXP15-C
+# Do not place a semicolon on the same line as an if, for or while statement
+def exp15(data):
+    for scope in data.scopes:
+        if scope.type == 'If' or scope.type == 'For' or scope.type == 'While':
+            token = scope.bodyStart.next 
+            if not token:
+                continue
+            if token.str==';':
```

Comment:
I believe there would be false positive here:

    if (x)
        ;

Reply:
Correct. I updated the check and test... should be good now